src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -122,12 +122,16 @@ internal unsafe SocketError DoOperationConnect(Socket socket, SafeCloseSocket ha
 
         internal SocketError DoOperationDisconnect(Socket socket, SafeCloseSocket handle)
         {
-            throw new PlatformNotSupportedException(SR.net_sockets_disconnect_notsupported);
+            SocketError socketError = SocketPal.Disconnect(socket, handle, _disconnectReuseSocket);
+            if (socketError != SocketError.IOPending)
+            {
+                FinishOperationSync(socketError, 0, SocketFlags.None);
+            }
+            return socketError;
         }
 
         private void InnerStartOperationDisconnect()
         {
-            throw new PlatformNotSupportedException(SR.net_sockets_disconnect_notsupported);
```

Comment:
If this resource isn't used any more, it should be deleted from the Strings.resx.